feat: dotnet standards tests + formatting pass on samples [MTT-5183]

fernando-cortez opened the pull request:
### Description
This PR adds the relevant Yamato job definition files, and dotnet project for creating code standards jobs per sample per commit when PRs target develop/main.

I had to include a code formatting pass to this PR such that the included tests would actually pass as well. So, there are a lot of files included, but it's all formatting changes, for at least the .cs files.

Files added are:
* .editorconfig file
* yml files inside .yamato/
* dotnet-tools subfolder

This and every subsequent PR will now produce the following code standard status checks in GitHub:
![image](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/assets/75813458/e2ee182e-0d37-4835-8c1f-c20e07774334)

Following the link of each code standard job in Yamato will reveal that each sample is scanned for whitespace + style/naming errors, all based on our repository's .editorconfig file's format preferences.
<!---
    Please provide a description of the changes proposed in the pull request.
    Make sure your commit messages have meaningful information.
    To help us link commits and PRs to JIRA work items, please include the JIRA ticket ID in the PR title or at least of your commit messages.
-->

### Issue Number(s)
[MTT-5183](https://jira.unity3d.com/browse/MTT-5183)
<!---
    Provide a list of fixed issues from Jira (MTT-ticketnumber) or GitHub (#issuenumber).
    This helps us understand the reasoning behind this change, what it fixes, feature being added, etc.
-->

### Contribution checklist
 - [ N/A ] Tests have been added for the project and/or any internal package
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

RikuTheFuffs commented on .editorconfig:
```diff
@@ -0,0 +1 @@
dotnet_naming_style.camel_case.required_prefix =
dotnet_naming_style.camel_case.required_suffix =
dotnet_naming_style.camel_case.word_separator =
dotnet_naming_style.camel_case.capitalization = camel_case
```
I se a lot of  `^M` characters in this file, is this, are these expected? What are they?

fernando-cortez replied:
> The ^M is a carriage-return character. If you see this, you're probably looking at a file that originated in the DOS/Windows world, where an end-of-line is marked by a carriage return/newline pair, whereas in the Unix world, end-of-line is marked by a single newline.

Per this [thread](https://unix.stackexchange.com/questions/32001/what-is-m-and-how-do-i-get-rid-of-it).

Ah, it seems like that file was originally pushed up with enforced Windows line endings (grabbed from Boss Room repo).
I've modified line endings on this file, so that a line ending is just a newline.
I did this locally, but I'll create a task to apply this to our repository's git settings!

RikuTheFuffs reviewed (commented):
Overall It's not really clear to me what this PR does. The JIRA ticket doesn't have any additional info.
Could you elaborate on what problem this is solving?

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=156) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=156) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=156) it.</sub>

fernando-cortez commented:
@RikuTheFuffs I added a bit more context to the description, and added to the description of the original issue (the duplicate contained info, not the original but that's sorted). Let me know if that helps!

RikuTheFuffs reviewed (approved):
Great, all good now!
